signer/src/transaction_coordinator.rs

```diff
@@ -2634,6 +2636,19 @@ pub fn assert_rotate_key_action(
     Ok((needs_verification, needs_rotate_key))
 }
 
+/// Adjust the wallet nonce based on the error
+pub fn adjust_nonce(wallet: &SignerWallet, error: &Error) {
+    match error {
+        // For `ConflictingNonceInMempool` we don't want to decrement the nonce
+        // to avoid failing also the following submissions
+        Error::StacksTxRejection(TxRejection {
+            reason: RejectionReason::ConflictingNonceInMempool,
+            ..
+        }) => (),
+        _ => wallet.set_nonce(wallet.get_nonce().saturating_sub(1)),
+    }
+}
+
```

Comment:
Maybe return from this function `Result<(), Error>` and return `Error::StacksTxRejection` to highlight that we actually didn't change anything?

Reply:
mmm it seems a bit strange to return an error from this, since we are doing some logic on the specific `StacksTxRejection` `ConflictingNonceInMempool` error. Like, the caller already received and handled the error, we are operating on that error and both outcomes (decrement or not) are "happy" paths for this fn. Returning an error would signal that something went wrong and we may need to act on it, while it's not really the case imho.

Reply:
Also not related to this pr but it is sad a bit that we mutate wallet by immutable reference =(